include/lexy/dsl/char_class.hpp

```diff
@@ -209,6 +209,21 @@ struct char_class_base : token_base<Derived>, _char_class_base
             {
                 return false;
             }
+            else if constexpr (std::is_same_v<typename Reader::encoding, lexy::default_encoding> ||
+                               std::is_same_v<typename Reader::encoding, lexy::byte_encoding>)
+            {
+                if (reader.peek() == Reader::encoding::eof())
+                    return false;
+
+                auto cp = reader.peek();
+                reader.bump();
+
+                if (!Derived::char_class_match_cp(cp))
```

Comment:
Please add a `static_cast` to char32_t here (CI is going to complain about that anyway).